packages/blockchain-wallet-v4-frontend/src/data/components/debitCard/sagas.test.js

```diff
@@ -0,0 +1,142 @@
+      const expected = true
+
+      expect(actual).toEqual(expected)
+    }
+
+    describe('success flow with currentAccount data', () => {
+      const generator = saga(CARD_ID)
+
+      it('should yields select getEligibleAccounts', () => {
+        validateYieldSelectGetEligibleAccounts(generator)
+      })
+
+      it('should yields put getCurrentCardAccountLoading', () => {
+        validateYieldPutGetCurrentCardAccountLoading(generator)
+      })
+
+      it('should yield call getDCCurrentAccount', () => {
+        validateYieldCallGetDCCurrentAccount(generator)
+      })
+
+      it('should yield put getCurrentCardAccountSuccess', () => {
+        const actual = generator.next(currentAccountSelectedMock).value
+        const expected = put(A.getCurrentCardAccountSuccess(accountMock))
+
+        expect(actual).toEqual(expected)
+      })
+
+      it('should be done', () => {
+        validateShouldBeDone(generator)
+      })
+    })
+
+    describe('no current account obtained from API', () => {
+      const generator = saga(CARD_ID)
+
+      it('should yields select getEligibleAccounts', () => {
+        validateYieldSelectGetEligibleAccounts(generator)
+      })
+
+      it('should yields put getCurrentCardAccountLoading', () => {
+        validateYieldPutGetCurrentCardAccountLoading(generator)
+      })
+
+      it('should yield call getDCCurrentAccount', () => {
+        validateYieldCallGetDCCurrentAccount(generator)
+      })
+
+      it('should yield put getCurrentCardAccountFailure', () => {
+        const noCurrentAccountResponse = { accountCurrency: null }
+        const actual = generator.next(noCurrentAccountResponse).value
+        const expected = put(
+          A.getCurrentCardAccountFailure('Could not get current user funds, no_funds_obtained')
+        )
+
+        expect(actual).toEqual(expected)
+      })
+
+      it('should be done', () => {
+        validateShouldBeDone(generator)
+      })
+    })
+
+    describe('current account mismatch with existing eligible accounts', () => {
+      const generator = saga(CARD_ID)
+
+      it('should yields select getEligibleAccounts', () => {
+        validateYieldSelectGetEligibleAccounts(generator)
+      })
+
+      it('should yields put getCurrentCardAccountLoading', () => {
+        validateYieldPutGetCurrentCardAccountLoading(generator)
+      })
+
+      it('should yield call getDCCurrentAccount', () => {
+        validateYieldCallGetDCCurrentAccount(generator)
+      })
+
+      it('should yield put getCurrentCardAccountFailure', () => {
+        const noCurrentAccountResponse = { accountCurrency: 'ARS' }
+        const actual = generator.next(noCurrentAccountResponse).value
+        const expected = put(
+          A.getCurrentCardAccountFailure('Could not get current user funds, no_funds_obtained')
+        )
+
+        expect(actual).toEqual(expected)
+      })
+
+      it('should be done', () => {
+        validateShouldBeDone(generator)
+      })
+    })
+  })
+})
```

Comment:
no worries, good to know, maybe to add it to our team documents about best practice since you spent some decent time on it, not to play around with ts in such a case 🙏